### DISCUSSION TITLE
MediaComposition.Add of a BackgroundAudioTrack - can I fully preserve its quality?

### DISCUSSION THREAD

**Discussion by jtorjo** [microsoft/WindowsAppSDK#1301] [Q&A] [answered]
I have this code:

	var composition = new MediaComposition();
	composition.Clips.Add(clip);

	var mp3 = await this.music_file();
	if (mp3 != "") {
		var music_file = await StorageFile.GetFileFromPathAsync(mp3);
		var music = await BackgroundAudioTrack.CreateFromFileAsync(music_file);
		music.TrimTimeFromStart = TimeSpan.Zero;
		music.TrimTimeFromEnd = TimeSpan.Zero;
		composition.BackgroundAudioTracks.Add(music);
	}

The problem is that in the final .mp4, quite a bit of the audio quality is lost. My .mp3 contains a lot of sound effects very close to each other. My mp3 sounds perfect, but when added to the video, the final video is quite worse in **audio quality**.

Is there a way to fully preserve the audio quality of the original audio?

**Comment by btueffers** [comment, marked as answer]
Hi @jtorjo, this topic would be more appropriate for the [Microsoft Q&A forum](https://docs.microsoft.com/answers/topics/windows-uwp.html) since those APIs are in the realm of Windows Platform APIs
